Update readme

savq opened the pull request:
- Explain features in a list
- Add requirements and installation sections
- Add section headers
- Check spelling
- Update terminology (nvimrc -> init.vim)
- Add link to pynvim

savq committed:
Update readme

- Explain features in a list
- Add requirements and installation sections
- Add section headers
- Check spelling
- Update terminology (nvimrc -> init.vim)

savq committed:
Update readme

- Add link to pynvim
- Remove mention of package manager in usage section
  (already explained in installation)
- minor fixes

smolck commented on README.md:
```diff
@@ -1,10 +1,36 @@
+## Requirements
+
+- Julia 1.0
+- Neovim 0.4 <!-- FIXME: is this right? -->
```
I'm not sure what the earliest version of Neovim is that this library supports (perhaps @bfredl does?), but at the very least I think this should be `0.4 or later` instead of just `0.4` (or whichever version is the earliest usable), so that it's clear this library is usable on later versions of Neovim/Neovim master.

Also, perhaps the earlier statement `Currently it assumes nvim is in $PATH` could be consolidated into this line, e.g. `Neovim 0.4 or later (this library assumes it is in $PATH)` or something like that.

smolck commented on README.md:
```diff
@@ -1,10 +1,36 @@
+
+### Using as an embedded process
+
+The Simplest way to test the API client is to spawn an embedded instance:
```
`Simplest` -> `simplest`

smolck commented on README.md:
```diff
@@ -17,7 +43,7 @@ nvim = nvim_connect("/socket/address")
 As a shortcut, `nvim = nvim_env()` will use the address in `$NVIM_LISTEN_ADDRESS`. This is useful to connect to the "parent" nvim instance when running the Julia REPL in a nvim terminal window.
 
-All API methods defined in metadata is defined as corresponding julia functions on the `Neovim` module, except that the `vim_`/`buffer_` prefix is dropped (as the reciever type is identified by the first argument anyway), except for `vim_eval` as `eval` is not overloadable. For instance:
+All API methods defined in metadata is defined as corresponding Julia functions on the `Neovim` module, except that the `vim_`/`buffer_` prefix is dropped (as the receiver type is identified by the first argument anyway), except for `vim_eval` as `eval` is not overloadable. For instance:
```
Can also correct the singular verb, so it should be, "All API methods defined in metadata *are* defined . . . ."

May also want to specify/elaborate on what "defined in metadata" means here, since that's not very clear from this paragraph; I think it's simply referring to the API functions that Neovim defines?

savq replied:
Yeah, it refers to the remote plugins API. There's a comment in the source suggesting that these functions should be inside another module (`Neovim.API`), which sounds like a good idea, since you could bring them to scope like `import Neovim.API as api`.

I'm also not sure about removing the `vim_` prefix. I get that it's kind of redundant with the methods signatures, but I think consistency with the API docs is more important. I'll see if I open an issue about this.

smolck replied:
> I'm also not sure about removing the vim_ prefix. I get that it's kind of redundant with the methods signatures, but I think consistency with the API docs is more important. I'll see if I open an issue about this.

Yeah; perhaps it would make more sense if the methods were defined on the structs themselves, so that you could do this sort of thing: `nvim.get_buffers`, `buf.set_line`, etc. That's what I've done in dart-nvim-api and seen in other nvim API bindings (e.g. nvim-rs), and it seems to be the cleanest way in languages with classes/structs/etc.

smolck commented on README.md:
```diff
@@ -1,10 +1,36 @@
+
+## Usage
+
+### Using as an embedded process
```
Just a thought, perhaps the `Using as` part of these subheadings can be dropped, since they're under the `Usage` heading? E.g. this would be `As an embedded process`, and the one lower down `As a plugin host`.

smolck commented on README.md:
````diff
@@ -27,13 +53,16 @@ set_line(buf, 1, "some text")
+```
+set rtp+=~/.julia/packages/Neovim/
+```
+A Julia plugin can then be defined in a `rplugin/julia/` subdirectory to your nvim directory or to a plugin repo. Functions defined at toplevel can be exported using macros in the Neovim module. `@fn`, `@command`, `@autocmd` can be used, as well as variants ending with `sync`.
````
May want to add a parenthetical about what "your nvim directory" usually is (`$XDG_CONFIG_HOME/.config/nvim`)?

Also, may also want to add some text there at the end like, "Here's an example:" so that it flows better into the example code underneath.

smolck reviewed (changes requested):
Thanks for the PR! LGTM for the most part, just left some small suggestions/comments.

savq committed:
Update readme (review)

See #14

- Clarify requirements' versions
- Minor fixes
- Add more references to nvim's docs

savq commented:
I agree with everything. made another commit.

The package needs more documentation around some places, but I think this is a good start.

smolck reviewed (approved):
LGTM! Thanks again for the PR; I'll go ahead and merge this and any further improvements can be done in subsequent PRs.
